kubernetes: add sysext documentation (#338)
flatcar-linux/flatcar-docs: docs/container-runtimes/getting-started-with-kubernetes.md

## Using Kubeadm

`kubeadm` remains one standard way to quickly deploy and operate a Kubernetes cluster. It's possible to install the tools (`kubeadm`, `kubelet`, etc.) using Ignition or directly with the Kubernetes sysext image distributed from the [flatcar/sysext-bakery][sysext-bakery] release page.

### Setup the control plane

> Here's an example with [butane][butane] to setup a control plane using the Systemd sysext approach or the binaries approach.

Reviewer: ```suggestion
Here are two examples to setup a control plane with [Butane][butane]. The first example is using the systemd-sysext approach to bring in the binaries and update them through systemd-sysupdate. The second approach fetches the binaries but has no way of updating them in-place.
```

Reviewer: The changes are gone because the addition of `/etc/sysupdate.kubernetes.d/kubernetes.conf` and the force push flushed them^^ https://github.com/flatcar/flatcar-docs/compare/2e44a5af9434779c7172b276b9fd4f4fcd24a398..a3ec8195b004ced01d8b397d82e1400a09b7bcc2

Reviewer: Maybe you can get the commit from the reflog or from github :D

<div>
  <ul class="nav nav-tabs">

>     <li class="active"><a href="#sysext" data-toggle="tab">With Systemd sysext</a></li>
>     <li><a href="#no-sysext" data-toggle="tab">Without Systemd sysext</a></li>

Reviewer: ```suggestion
    <li class="active"><a href="#sysext" data-toggle="tab">With systemd-sysext</a></li>
    <li><a href="#no-sysext" data-toggle="tab">Without systemd-sysext</a></li>
```

  </ul>
  <div class="tab-content coreos-docs-image-table">
    <div class="tab-pane" id="sysext">
      <div class="channel-info">

>         This is an example using Systemd sysext and Systemd sysupdate:

Reviewer: ```suggestion
        This is an example using systemd-sysext and systemd-sysupdate:
```

        <pre>
---
version: 1.0.0
variant: flatcar
storage:
  links:
    - target: /opt/extensions/kubernetes/kubernetes-v1.27.4-x86-64.raw
      path: /etc/extensions/kubernetes.raw
      hard: false
  files:
    - path: /etc/sysupdate.d/noop.conf
      contents:
        source: https://github.com/flatcar/sysext-bakery/releases/download/20230901/noop.conf
    - path: /opt/extensions/kubernetes/kubernetes-v1.27.4-x86-64.raw
      contents:
        source: https://github.com/flatcar/sysext-bakery/releases/download/20230901/kubernetes-v1.27.4-x86-64.raw
systemd:
  units:
    - name: systemd-sysupdate.timer
      enabled: true
    - name: systemd-sysupdate.service
      dropins:
        - name: kubernetes.conf
          contents: |
            [Service]

>             ExecStartPre=/usr/lib/systemd/systemd-sysupdate -C kubernetes update

Reviewer: Should we run refresh here?

Reviewer: Thanks, that's good :) Any tweaks to how that is done can also be added later.

> Here's the [butane][butane] configuration to setup the nodes.

Reviewer: ```suggestion
Here's the [butane][butane] configuration to setup the nodes with plain binaries (no way to easily update).
```

<div>
  <ul class="nav nav-tabs">

>     <li class="active"><a href="#sysext" data-toggle="tab">With Systemd sysext</a></li>
>     <li><a href="#no-sysext" data-toggle="tab">Without Systemd sysext</a></li>

Reviewer: ```suggestion
    <li><a href="#sysext" data-toggle="tab">With systemd-sysext and updates</a></li>
    <li class="active"><a href="#no-sysext" data-toggle="tab">With plain binaries and no updates</a></li>
```

  </ul>
  <div class="tab-content coreos-docs-image-table">
    <div class="tab-pane" id="sysext">
      <div class="channel-info">

>         This is an example using Systemd sysext and Systemd sysupdate:

Reviewer: ```suggestion
        This is an example using systemd-sysext and systemd-sysupdate:
```

        <pre>
---
version: 1.0.0
variant: flatcar
storage:
  links:
    - target: /opt/extensions/kubernetes/kubernetes-v1.27.4-x86-64.raw
      path: /etc/extensions/kubernetes.raw
      hard: false
  files:
    - path: /etc/sysupdate.d/noop.conf
      contents:
        source: https://github.com/flatcar/sysext-bakery/releases/download/20230901/noop.conf
    - path: /opt/extensions/kubernetes/kubernetes-v1.27.4-x86-64.raw
      contents:
        source: https://github.com/flatcar/sysext-bakery/releases/download/20230901/kubernetes-v1.27.4-x86-64.raw
systemd:
  units:
    - name: systemd-sysupdate.timer
      enabled: true
    - name: systemd-sysupdate.service
      dropins:
        - name: kubernetes.conf
          contents: |
            [Service]

>             ExecStartPre=/usr/lib/systemd/systemd-sysupdate -C kubernetes update

Reviewer: Here, too, should we run refresh?
